### PR TITLE
fix(core): use fd-button for carousel button and fix focus after last item reached

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#3880

#### Please provide a brief summary of this pull request.
Earlier importing ButtonModule in carousel module was having issue. which got resolved with new structure of main branch.
So, now using fd-button for carousel button and removed un-necessary code present.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [NA] update `README.md`
- [NA] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [NA] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/carousel/carousel.component.html [hunk 1]
```diff
@@ -66,17 +66,15 @@
 </ng-template>
 
 <ng-template #buttonLeft>
-    <button class="fd-button fd-carousel__button fd-carousel__button--left" [style.z-index]="1" data-slide="previous"
-            [attr.aria-label]="leftNavigationBtnLabel" (click)="previous(true); $event.stopPropagation();"
-            [disabled]="leftButtonDisabled" [attr.title]="leftNavigationBtnLabel">
-        <i [class]="_isRtl() ? 'sap-icon--slim-arrow-right' : 'sap-icon--slim-arrow-left'"></i>
-    </button>
+    <button fd-button class="fd-carousel__button fd-carousel__button--left" [style.z-index]="1" data-slide="previous"
+        [attr.aria-label]="leftNavigationBtnLabel" (click)="previous(true); $event.stopPropagation();"
```
Comment (platon-rov) [inline libs/core/src/lib/carousel/carousel.component.html:70]: Does the flag usage needed here? Same for the next method. Seems like it's always used with value `true`

File: libs/core/src/lib/carousel/carousel.component.html [hunk 2]
```diff
@@ -66,17 +66,15 @@
...
 </ng-template>
 
 <ng-template #buttonRight>
-    <button class="fd-button fd-carousel__button fd-carousel__button--right" data-slide="next"
-            [attr.aria-label]="rightNavigationBtnLabel" (click)="next(true); $event.stopPropagation();"
-            [disabled]="rightButtonDisabled" [attr.title]="rightNavigationBtnLabel">
-        <i [class]="_isRtl() ? 'sap-icon--slim-arrow-left' : 'sap-icon--slim-arrow-right'"></i>
-    </button>
+    <button fd-button class="fd-carousel__button fd-carousel__button--right" [style.z-index]="1" data-slide="next"
+        [attr.aria-label]="rightNavigationBtnLabel" (click)="next(true); $event.stopPropagation();"
```
Comment (platon-rov) [inline libs/core/src/lib/carousel/carousel.component.html:77]: Please check https://github.com/SAP/fundamental-ngx/pull/5586/files#r645516510

File: libs/core/src/lib/carousel/carousel.component.html (hunk 1 above)
Comment (DeepakSap14) [reply libs/core/src/lib/carousel/carousel.component.html]: after refactor, it is not needed. corrected the logic here, for the focus after last item is reached.
fixed now.

File: libs/core/src/lib/carousel/carousel.component.html (hunk 2 above)
Comment (DeepakSap14) [reply libs/core/src/lib/carousel/carousel.component.html]: fixed now.

**Comment by droshev** [comment]
Closing PRs - feel free to reopen it if/when needed

**Comment by DeepakSap14** [comment]
Reopening for review and approvals.

**Review by platon-rov — APPROVED** [review]

**Comment by DeepakSap14** [comment]
have a minor bug in keyboard navigation, fixing that. will push changes in few minutes.

**Review by kavya-b — APPROVED** [review]
LGTM

**Comment by maxym-valor** [comment]
Approved!

**Review by Betrozov — APPROVED** [review]

**Pull request merged**